### PR TITLE
Progressbar

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This can be used from  the notebook to do some simple progress information.

cc @rgbkrk

### CODE REVIEW

File: IPython/core/display.py [hunk 1]
```diff
@@ -728,6 +728,38 @@ def data(self, svg):
...
+class ProgressBar(DisplayObject):
+    """Progressbar supports displaying a progressbar like element 
+    """
+    def __init__(self, total):
+        """Creates a new progressbar
+        
+        Parameters
+        ----------
+        total : int
+            maximnum size of the progressbar
```
Comment (rgbkrk) [inline IPython/core/display.py]: One lil' typo: maximum

**Comment by rgbkrk** [comment]
An example notebook would be great to have as well.

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Can you add this to the whatsnew collection in here: https://github.com/ipython/ipython/tree/master/docs/source/whatsnew/pr

**Comment by rgbkrk** [comment]
@meseeksdev backport

**Comment by rgbkrk** [comment]
@meeseeksdev backport

**Comment by rgbkrk** [comment]
No.

**Comment by rgbkrk** [comment]
@meeseeksdev backport

**Comment by rgbkrk** [comment]
:wink:

/cc @Carreau

**Comment by rgbkrk** [comment]
@meeseeksdev coffee please

**Comment by rgbkrk** [comment]
Thanks @mariusvniekerk

**Comment by Carreau** [comment]
hum.. we should likely try to make it work better in the terminal as it does not erase the previous version of the progressbar, leading to a progress Triangle:

```
In [6]: pb = ProgressBar(100)
   ...: for i in range(25):
   ...:     pb.progress = i*4
   ...:
   ...:
[                                                            ] 0/100
[==                                                          ] 4/100
[====                                                        ] 8/100
[=======                                                     ] 12/100
[=========                                                   ] 16/100
[============                                                ] 20/100
[==============                                              ] 24/100
[================                                            ] 28/100
[===================                                         ] 32/100
[=====================                                       ] 36/100
[========================                                    ] 40/100
[==========================                                  ] 44/100
[============================                                ] 48/100
[===============================                             ] 52/100
[=================================                           ] 56/100
[====================================                        ] 60/100
[======================================                      ] 64/100
[========================================                    ] 68/100
[===========================================                 ] 72/100
[=============================================               ] 76/100
[================================================            ] 80/100
[==================================================          ] 84/100
[====================================================        ] 88/100
[=======================================================     ] 92/100
[=========================================================   ] 96/100
```

**Comment by rgbkrk** [comment]
Oh hey, we better stick some `\r` in there.

**Comment by Carreau** [comment]
Yeah, it's a bit more complicated as we need to have the first _repr_ not
have a `\r`... and its due to `update_display` messages, so if something
was displayed in the meantime, it wont' work :-9

**Comment by Carreau** [comment]
And we need the  "print" to have (end=''), except for the last one...

**Comment by takluyver** [comment]
And if something else is displayed interleaved, printing with `end=''` will mean it's further messed up, unfortunately.

I think it might work to print with `end='\r'` - it shouldn't clear the line until more stuff is printed to overwrite it. Not 100% sure of that, though

**Pull request merged**